[DDP] remove duplicate var check in reducer

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#53919 [DDP]  remove dedupe check in reducer**

https://github.com/pytorch/pytorch/pull/53279/files has landed
deduplicating the shared params in python before constructing reducer. Because
of this, we no longer need the changes in
https://github.com/pytorch/pytorch/pull/46755/files.

This is already tested by `test_ddp_shared_grad_acc_unused_params` and
`test_ddp_weight_sharing`

Differential Revision: [D27015466](https://our.internmc.facebook.com/intern/diff/D27015466/)

rohan-varma committed:
[DDP] undo changes in 46755

https://github.com/pytorch/pytorch/pull/53279/files has landed
deduplicating the shared params in python before constructing reducer. Because
of this, we no longer need the changes in
https://github.com/pytorch/pytorch/pull/46755/files.

This is already tested by `test_ddp_shared_grad_acc_unused_params` and
`test_ddp_weight_sharing`

Differential Revision: [D27015466](https://our.internmc.facebook.com/intern/diff/D27015466/)

[ghstack-poisoned]

wayi1 reviewed (approved):
Thanks for the cleanup!

rohan-varma committed:
Update on "[DDP] undo changes in 46755"


https://github.com/pytorch/pytorch/pull/53279/files has landed
deduplicating the shared params in python before constructing reducer. Because
of this, we no longer need the changes in
https://github.com/pytorch/pytorch/pull/46755/files.

This is already tested by `test_ddp_shared_grad_acc_unused_params` and
`test_ddp_weight_sharing`

Differential Revision: [D27015466](https://our.internmc.facebook.com/intern/diff/D27015466/)

[ghstack-poisoned]
